puzzle redesign epic

stormcloud266 opened the pull request:
- setup db, update seed file and sdls
- remove step thumbnails and style carousel
- style pagination buttons
- remove global x padding. make carousel responsive
- update other pages' wrappers

stormcloud266 committed:
setup db, update seed file and sdls

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

stormcloud266 committed:
remove step thumbnails and style carousel

stormcloud266 committed:
style pagination buttons

stormcloud266 committed:
remove global x padding. make carousel responsive

stormcloud266 committed:
update other pages' wrappers

stormcloud266 committed:
temporary wrapper fix for puzzle landing layout

stormcloud266 committed:
more wrapper fixes

stormcloud266 committed:
move requirements to puzzle

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

stormcloud266 committed:
add cover image to puzzle. fix step content layout

stormcloud266 committed:
style modal in carousel

stormcloud266 committed:
update schema and seed

stormcloud266 committed:
regenerate sdls/services

stormcloud266 committed:
bring pages into steps layout

stormcloud266 committed:
update carousel styles

stormcloud266 committed:
style carousel, add solution image to schema

stormcloud266 committed:
fix carousel dots position. add solution page

stormcloud266 committed:
add stepPageLayout component

stormcloud266 committed:
update step category to guideType

stormcloud266 committed:
take in overlay content

stormcloud266 committed:
add comments, small refactor

stormcloud266 committed:
consolidate schema

stormcloud266 committed:
remove console log

IvySwing committed:
itsmekori/eng-980-import-all-overlay-svgs (#968)

* create components for overlay icons

* update file name and removed imports

stormcloud266 committed:
Revert "itsmekori/eng-980-import-all-overlay-svgs" (#969)

Revert "itsmekori/eng-980-import-all-overlay-svgs (#968)"

This reverts commit dc6166ccbf95d9492003faa2bfdd66195c3e284c.

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

illepic commented on api/db/schema.prisma:
```diff
@@ -198,9 +210,43 @@ model Step {
+  /// If this is not provided, the solution page will use the default image
+  solutionImage String?
+  stepPage      StepPage[]
+  stepGuideType StepGuideType
```
Throw in a comment about what the heck this is.

illepic commented on api/db/schema.prisma:
```diff
@@ -198,9 +210,43 @@ model Step {
+  @@unique([stepId, sortWeight(sort: Asc)])
+}
+
+enum StepGuideType {
```
Comment about what do.

illepic commented on web/src/components/MarkdownCarousel/MarkdownCarousel.tsx:
```diff
@@ -1,34 +1,63 @@
+        containerClassName: 'relative -bottom-12',
       }}
-      className="pb-14"
+      className="!h-full w-full"
```
Let's try https://formidable.com/open-source/nuka-carousel/api#style for styles on parent.

illepic commented on web/src/components/RewardablesGrid/RewardablesGrid.tsx:
```diff
@@ -14,6 +14,8 @@ import Thumbnail from 'src/components/Thumbnail/Thumbnail'
 import { rewardableLandingRoute } from 'src/lib/urlBuilders'
 import { GridLandingRouteType } from 'src/lib/urlBuilders'
 
+import Wrapper from '../Wrapper/Wrapper'
```
Do the `src/path` here.

illepic commented on web/src/components/RewardablesGrid/RewardablesGrid.tsx:
```diff
@@ -61,80 +63,19 @@ const RewardablesList = ({
+              className={clsx(
+                'grid grid-cols-1 gap-6 sm:grid-cols-2',
+                'md:grid-cols-3 lg:grid-cols-4'
+              )}
```
Let's just remove `clsx` and make both those class strings a single string.

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

stormcloud266 committed:
added schema comments, minor style fixes

stormcloud266 committed:
Itsmekori/eng 980 import all overlay svgs (#970)

* create components for overlay icons

* update file name and removed imports

---------

Co-authored-by: IvySwing <entense@att.net>

stormcloud266 committed:
ENG-981: remove anonymous puzzles concept (#971)

* remove anon references

* remove reconciliation code

* remove unused vars

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

stormcloud266 committed:
ENG-982: add new puzzle landing concept (#973)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

IvySwing committed:
Itsmekori/eng 984 dynamic overlay content (#976)

* create components for overlay icons

* update file name and removed imports

* updated component name and added obbect lookup

* updated code to make overlay content dynamic

---------

Co-authored-by: Tawnee <tawnee.a@proton.me>

stormcloud266 committed:
ENG-985: add step landing page (#975)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* fix linting errors

* skip tests

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

stormcloud266 committed:
ENG-986: Login modal to context (#974)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

---------

Co-authored-by: Tawnee <allprt89@gmail.com>
Co-authored-by: Christopher Bloom <bloomcb@gmail.com>

stormcloud266 committed:
ENG-988: Start quest button (#979)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* quest button navigates to step page. add href to button

---------

Co-authored-by: Tawnee <allprt89@gmail.com>
Co-authored-by: Christopher Bloom <bloomcb@gmail.com>

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

IvySwing committed:
itsmekori/eng-987-update-simple-text-input (#977)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* remove lock icon & submit button, and update text

* updated css for new UI design

* updated styling on input and display value

* update styling on button component

* update styling on hover state and added colors

* updates to styling

* update shadow style

---------

Co-authored-by: Tawnee <allprt89@gmail.com>
Co-authored-by: Tawnee <tawnee.a@proton.me>

IvySwing committed:
Itsmekori/eng 997 responsive overlay icons (#990)

* remove anon references

* remove reconciliation code

* remove unused vars

* itsmekori/eng-980-import-all-overlay-svgs (#968)

* create components for overlay icons

* update file name and removed imports

* Revert "itsmekori/eng-980-import-all-overlay-svgs" (#969)

Revert "itsmekori/eng-980-import-all-overlay-svgs (#968)"

This reverts commit dc6166ccbf95d9492003faa2bfdd66195c3e284c.

* popup-updates (#962)

Add play button. update styles

Co-authored-by: Tawnee <allprt89@gmail.com>

* save external address (#972)

* add external address to schema

* update/display external wallet

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* remove test puzzle constant and update references (#965)

Co-authored-by: Tawnee <allprt89@gmail.com>

* added schema comments, minor style fixes

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* skip tests

* Itsmekori/eng 980 import all overlay svgs (#970)

* create components for overlay icons

* update file name and removed imports

---------

Co-authored-by: IvySwing <entense@att.net>

* ENG-981: remove anonymous puzzles concept (#971)

* remove anon references

* remove reconciliation code

* remove unused vars

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* Itsmekori/eng 984 dynamic overlay content (#976)

* create components for overlay icons

* update file name and removed imports

* updated component name and added obbect lookup

* updated code to make overlay content dynamic

---------

Co-authored-by: Tawnee <tawnee.a@proton.me>

* remove lock icon & submit button, and update text

* updated css for new UI design

* updated styling on input and display value

* update styling on button component

* update styling on hover state and added colors

* updates to styling

* update shadow style

* added mini icons to existing components

* update classnames, imports & max width dimensions

* Proper stepBySlug query

* update components & add components for mini icons

* setting up toggle between larger and mini svgs

---------

Co-authored-by: Tawnee <allprt89@gmail.com>
Co-authored-by: Tawnee <tawnee.a@proton.me>
Co-authored-by: Christopher Bloom <bloomcb@gmail.com>

stormcloud266 committed:
ENG-990: step success page (#984)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* quest button navigates to step page. add href to button

* refetch query on successful makeAttempt

* Use correct icons for step state. Block users from accessing steps

* just pass refetch function. use success messages

* cleanup remaining steps text

* minor UI updates (#985)

* update seed to use nfts i have

* minor style updates

* rename modal to overlay

* update step progress icons and styles

* update step progress text position

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* merge with epic

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

stormcloud266 committed:
remove puzzle id (#978)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* remove puzzle id

* remove leftover type

* fix graphql query

* return div if step not found

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

stormcloud266 committed:
ENG-998: connect make attempt to new simple text input (#992)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* quest button navigates to step page. add href to button

* refetch query on successful makeAttempt

* Use correct icons for step state. Block users from accessing steps

* just pass refetch function. use success messages

* cleanup remaining steps text

* minor UI updates (#985)

* update seed to use nfts i have

* minor style updates

* rename modal to overlay

* update step progress icons and styles

* update step progress text position

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* minor style updates. use makeAttempt in simple text

* fix errorMessage

* ENG-999: success status on puzzle landing (#993)

* show success message for completed puzzles

* update puzzle jsx

* ENG-1000: randomize successfail messages (#995)

* hardcode success and fail messages for steps

* remove fail message from other step types

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* usecallback in simple text input

* use template literal

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

IvySwing committed:
itsmekori/eng-1004-bring-in-puzzle-landing-icons (#999)

created & added puzzles svg icons

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

stormcloud266 committed:
ENG-994: replace get started icons and content (#1000)

* replace get started placeholder content

* Remove bg-blacks

---------

Co-authored-by: Tawnee <allprt89@gmail.com>
Co-authored-by: Christopher Bloom <bloomcb@gmail.com>

stormcloud266 committed:
ENG-983: update step/puzzle header to new ui (#1001)

* remove anon references

* remove reconciliation code

* remove unused vars

* remove steps from puzzle page. style puzzle landing

* handle unauthenticated users in puzzle landing

* get step by slug, remove optionStep and stepsCell

* update step cell types

* rename current login to form and use modal for transition, update references

* fix linting errors

* fix linting errors

* skip tests

* quest button navigates to step page. add href to button

* refetch query on successful makeAttempt

* Use correct icons for step state. Block users from accessing steps

* just pass refetch function. use success messages

* cleanup remaining steps text

* minor UI updates (#985)

* update seed to use nfts i have

* minor style updates

* rename modal to overlay

* update step progress icons and styles

* update step progress text position

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* minor style updates. use makeAttempt in simple text

* fix errorMessage

* ENG-999: success status on puzzle landing (#993)

* show success message for completed puzzles

* update puzzle jsx

* ENG-1000: randomize successfail messages (#995)

* hardcode success and fail messages for steps

* remove fail message from other step types

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

* usecallback in simple text input

* use template literal

* add context, use rewardable name in nav

* revert to previous profile button

* use variables

* remove step check

* update heading font size

* remove usememo

---------

Co-authored-by: Tawnee <allprt89@gmail.com>

stormcloud266 committed:
Merge branch 'main' into puzzle-redesign-epic

stormcloud266 committed:
update node v (#1002)

Co-authored-by: Tawnee <allprt89@gmail.com>
